incubator/hnc/internal/reconcilers/object.go

```diff
@@ -359,9 +360,20 @@ func (r *ObjectReconciler) syncPropagated(ctx context.Context, log logr.Logger,
 	if srcInst == nil {
 		return actionRemove, nil
 	}
-
 	// If an object doesn't exist, assume it's been deleted or not yet created.
 	exists := inst.GetCreationTimestamp() != v1.Time{}
+	selector := r.getSelector(ctx, log, srcInst)
+
+	instLabels := labels.Set(r.Forest.Get(inst.GetNamespace()).GetLabels())
```

Comment:
This is not the `instLabels`, it's the namespace's labels. So let's call this `nsLabels`.

Reply:
done